url-router/src/main/java/com/bukalapak/urlrouter/ParamMap.kt

```diff
@@ -16,50 +16,92 @@ open class ParamMap : HashMap<String, List<String>>() {
         return super.put(key, Arrays.asList(value))
     }
 
-    fun getString(key: String): String = if (containsKey(key)) {
+    fun getString(key: String): String? = super.get(key)?.firstOrNull()
+
+    fun getLong(key: String): Long? {
+        return try {
+            super.get(key)?.firstOrNull()?.toLong()
+        } catch (ex: Exception) {
+            0L
+        }
+    }
+
+    fun getInt(key: String): Int? {
+        return try {
+            super.get(key)?.firstOrNull()?.toInt()
+        } catch (ex: Exception) {
+            0
+        }
+    }
+
+    fun getFloat(key: String): Float? {
+        return try {
+            super.get(key)?.firstOrNull()?.toFloat()
+        } catch (ex: Exception) {
+            0.0F
+        }
+    }
+
+    fun getDouble(key: String): Double? {
+        return try {
+            super.get(key)?.firstOrNull()?.toDouble()
+        } catch (ex: Exception) {
+            0.0
+        }
+    }
+
+    fun getBoolean(key: String): Boolean? = if (containsKey(key)) {
+        val value = super.get(key)?.firstOrNull()
+        value.equals("true", true) && value.equals("1", true)
```

Comment:
ini jangan `jika equals true`, tapi `jika not equals false`, sy liat dari kode url parser android, mngkin good practicenya (?)

Reply:
itu juga harusnya operator `||` kalo pake jika true atau `&&` jika operator jika bukan false

Reply:
kalau not equals false default value nya true mas
kalau value nya bukan boolean returnnya jadi true

iya ya pake ||